src/settings/from_value.rs

```diff
@@ -1,76 +1,146 @@
+use std::collections::HashMap;
+
 use super::Value;
 use log::error;
 
 // Trait to allow for conversion from rmpv::Value to any other data type.
 // Note: Feel free to implement this trait for custom types in each subsystem.
 // The reverse conversion (MyType->Value) can be performed by implementing `From<MyType> for Value`
 pub trait ParseFromValue {
-    fn parse_from_value(&mut self, value: Value);
+    fn parse_from_value(&mut self, value: &Value) -> bool;
```

Comment:
Instead of returning an opaque `bool`, possibly it could be cleaner to use a direct `Result` with a custom error type or even just a `()` instead. This would also make the control flow of most of the implementors cleaner, since then they don't need to wrap the entire function body in an `if value.is_map() {` with the error case at the end or the like.

Reply:
Totally, thank you!